### ISSUE TITLE
Handling merge conflict of database

### ISSUE THREAD

**Issue by raghubetina** [firstdraft/appdev_template#111]
I've always worried about the tradeoff between:

 - Conceptual clarity of including the sqlite database in the repo and therefore sidestepping the messiness of having to rollback migrations before switching branches / being able to solve data integrity issues by jumping back to previous commits.
 - Merge conflicts caused by data differences in different branches.

Is this a potential solution?

https://stackoverflow.com/questions/15232000/git-ignore-files-during-merge

If so, we should consider including an appropriate `.gitattribute` file in all projects.

**Comment by raghubetina** [comment]
@jelaniwoods This should maybe go into `web_git` repo, but anyway. I think we should prioritize this and add a merge strategy for the `development.sqlite3` file.

**Comment by jelaniwoods** [comment]
@raghubetina when would a merge even occur in the student's git workflow?

**Comment by raghubetina** [comment]
@jelaniwoods Yes, I believe checking out other branches shouldn't cause any issues. (Although I feel like I have noticed some "Pending migration" situations when I jump back and forth between branches, which I was hoping to avoid by ungitignoring `development.sqlite3`; I haven't had time to try and reliably reproduce.)

It's true that we currently landed upon a merge-free workflow, with `git push heroku HEAD:master -f` in a single-person-only-deploying-"best"-branch situation. However in the past (e.g. on Very Best) and in the future, we'll want students to work in teams and therefore they will need to learn to merge.

**Comment by jelaniwoods** [comment]
@raghubetina okay that makes sense. I assume then a `merge` section needs to be added to web_git at some point after the core functionality is working at acceptable speeds and without errors.

**Comment by jelaniwoods** [comment]
@raghubetina according to the `git` documentation and [this answer](https://stackoverflow.com/a/49358531) we need to be update the `git` configuration to use the merge strategy, I can think of a couple ways to do this but wanted to get your opinion:

- Can run the git command in `bin/setup`
- Can run the git command in a separate script that only runs once per workspace
  - Can shell out command or use `git` gem somehow

These approaches seem _mostly_ equivalent to me. Am overlooking something?

**Comment by raghubetina** [comment]
@jelaniwoods In the past we relied on GitHub's built-in merging UI, and I was hoping to continue to do so; I don't relish the idea of building a UI for resolving conflicts, although I suppose it wouldn't be _too_ bad.

I would like students to

 - Push to GitHub
 - merge their branch to `master` on GitHub (resolving conflicts there)
 - switch to `master` locally
 - pull from GitHub
 - checkout a new branch to work on their next task
 - rinse and repeat

I thought that defining a `.gitattributes` file and specifying a strategy for handling `development.sqlite3` might work while merging on GitHub, but now that I read that SO thread in more detail, perhaps GitHub won't be able to do it since we can't define the actual strategy on GitHub. Requires experimentation.

**Comment by jelaniwoods** [comment]
OH, okay. I will investigate with GitHub's merge to see if if will accept the strategy.

**Comment by jelaniwoods** [comment]
@raghubetina after a little research and testing, it doesn't appear that GItHub respects `.gitattributes` file.

This SO answer further suggests that GitHub doesn't look at the file https://stackoverflow.com/a/24382933/10481804
I tested in this PR https://github.com/jelaniwoods-FurtherLearning/wweeeee/pull/8

**Comment by raghubetina** [comment]
@jelaniwoods What happens when you have conflicts in a binary file during GitHub merge? Does it let you pick one or the other, or does it just fail?

**Comment by jelaniwoods** [comment]
In the PR I tested with the conflict of `development.sqlite3` couldn't be resolved in the web editor.

<img width="681" alt="Screen Shot 2020-06-09 at 3 46 11 PM" src="https://user-images.githubusercontent.com/17581658/84197906-5b1a6a80-aa68-11ea-97a5-d72acbb6dc3c.png">

**Comment by jelaniwoods** [comment]
@raghubetina If it's local merging we can use `.gitattributes` and pick a version so we don't need to resolve conflicts.

Trying to merge without `.gitattributes` looks tricky. There doesn't appear to be any UI change and even the diff on the command line is unhelpful.

<img width="658" alt="Screen Shot 2020-06-09 at 4 27 42 PM" src="https://user-images.githubusercontent.com/17581658/84202931-26a9ad00-aa6e-11ea-9172-b7fe0f74aafb.png">
